vllm/sequence.py

```diff
@@ -141,6 +140,18 @@ def __init__(
     def lora_int_id(self) -> int:
         return self.lora_request.lora_int_id if self.lora_request else 0
 
+    def hash(self, logical_idx: int) -> int:
+        # Compute the number of tokens in the sequence
+        num_tokens = logical_idx * self.block_size + self.block_size
+        return hash(tuple(self.data.get_token_ids()[0:num_tokens]))
+
+    def prefix_len_of_block(self, logical_idx: int, full_prefix_len: int):
+        num_tokens = logical_idx * self.block_size + self.block_size
+        if num_tokens > full_prefix_len:
+            return full_prefix_len
+        else:
+            return num_tokens
```

Comment:
Can you explain the logic here a bit? I'm a bit confused why we need a logic here to compute the prefix length. (for example, why don't directly use `full_prefix_len`)